clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -515,8 +518,13 @@ PointerVariableConstraint::PointerVariableConstraint(
   // Get a string representing the type without pointer and array indirection.
   BaseType = extractBaseType(D, TSInfo, QT, Ty, C);
 
-  IsVoidPtr = isTypeHasVoid(QT);
-  bool IsWild = !getIsGeneric() && (isVarArgType(BaseType) || IsVoidPtr);
+  // check if the type is some depth of pointers to void
+  // TODO: is this what the field should mean? do we want to include other
+  // indirection options like arrays?
+  IsVoidPtr = QT->isPointerType() && isTypeHasVoid(QT);
```

Comment:
Yeah, something about the condition isn't right
```c
void test() {
  void *arr[10];
}
```
converts to 
```c
void test() {
  void * arr[10] = {((void *)0)};
}
```
which is quite strange. It's not rewritten, but not to a checked type, even though looking at the `-debug-solver` output shows it solves to a checked type.